### PR TITLE
791 Fuel Grade Picker

### PR DESCRIPTION
**Pull request by thardie2**
### What this PR does :thinking: 
Fuel grade picker to 791 issue form

### Resolves :white_check_mark: 
[PBS-2183](https://jira.il2.dso.mil/browse/PBS-2183)

### Testing Items :microscope: 
- [ ] New Mission/Sortie should default to jp8
- [ ] Change fuel type and create new sortie
- [ ] New sortie should have changed fuel type

### CODE REVIEW

File: PBLogging/Views/SingleSortie/Form791/IssueLogView.swift [hunk 1]
```diff
@@ -7,13 +7,29 @@
...
     
     // MARK: - Properties
 
     @EnvironmentObject var dataController: DataController
     @ObservedObject var sortie: Sortie
     
+    init(sortie: Sortie) {
+        self.sortie = sortie
+        if self.sortie.form791.fuelGrade.isEmpty {
+            self.sortie.form791.fuelGrade = "JP8"
```
Comment (tsanders88) [inline PBLogging/Views/SingleSortie/Form791/IssueLogView.swift:29]: use enum here

File: PBLogging/Data/EntityExtensions/Sortie+Accessors.swift [hunk 2]
```diff
@@ -237,6 +237,14 @@ extension Sortie {
                     newForm791.refuelerNameAndGrade      = previousSortie.form791.refuelerNameAndGrade
                 }
                 
+                if let sorties = self.mission?.sorties {
+                    if sorties.count > 1 {
+                        let previousSortie = sorties[1]
```
Comment (tsanders88) [inline PBLogging/Data/EntityExtensions/Sortie+Accessors.swift:242]: is previous sortie always going to be sorties[1]?

File: PBLogging/Views/SingleSortie/Form791/IssueLogView.swift [hunk 3]
```diff
@@ -7,13 +7,29 @@
 
 import SwiftUI
 
+enum FuelTypes: String, CaseIterable {
```
Comment (tsanders88) [inline PBLogging/Views/SingleSortie/Form791/IssueLogView.swift:10]: probably makes sense to put in another file

File: PBLogging/Data/EntityExtensions/Sortie+Accessors.swift (hunk 2 above)
Comment (thardie2) [reply PBLogging/Data/EntityExtensions/Sortie+Accessors.swift:242]: For the most part, yes. If there are sorties without take off and landing times, the sorting gets a little random and couple potentially setting the fuel grade to something else but its random. When sorties have times entered, new sorties are index 0 and the previous is always 1. It was already setup to pull the fuel grade from the .last element of the array but if the grade changes mid mission, then new sorties will not have the new grade. Open to suggestions on a better way to do it.

Comment (tsanders88) [reply PBLogging/Data/EntityExtensions/Sortie+Accessors.swift:242]: it's probably fine, just wasn't expecting it to work that way i guess

**Review by tsanders88 — APPROVED** [review]
lgtm

**Pull request merged**